### PR TITLE
1.4 deprecation from docs

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Removes v1.4 from the version dropdown
Removes info between lastVersion="1.4" blocks
Removes firstVersion="1.4" blocks but leaves content
Removes some mentions of v1.4 from places where it's not mentioning a requirement

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 1]
```diff
@@ -301,9 +301,9 @@ models:
 
 ## Temporary tables
 
-Beginning in dbt version 1.3, incremental table merges for Snowflake prefer to utilize a `view` rather than a `temporary table`. The reasoning was to avoid the database write step that a temporary table would initiate and save compile time. 
+Incremental table merges for Snowflake prefer to utilize a `view` rather than a `temporary table`. The reasoning is to avoid the database write step that a temporary table would initiate and save compile time. 
 
-However, some situations remain where a temporary table would achieve results faster or more safely. dbt v1.4 adds the `tmp_relation_type` configuration to allow you to opt in to temporary tables for incremental builds. This is defined as part of the model configuration. 
+However, some situations remain where a temporary table would achieve results faster or more safely. The `tmp_relation_type` configuration allowa you to opt in to temporary tables for incremental builds. This is defined as part of the model configuration.
```
Comment (nghi-ly) [inline website/docs/reference/resource-configs/snowflake-configs.md]:
```suggestion
However, some situations remain where a temporary table would achieve results faster or more safely. The `tmp_relation_type` configuration enables you to opt in to temporary tables for incremental builds. This is defined as part of the model configuration. 
```

File: website/docs/reference/events-logging.md [hunk 2]
```diff
@@ -119,12 +117,10 @@ Many events are fired while compiling or running a specific DAG node (model, see
 }
 ```
 
-</VersionBlock>
-
 ## Python interface
 
 :::warning
-The `EVENT_HISTORY` object has been deprecated and removed in dbt Core v1.4+
+The `EVENT_HISTORY` object has been deprecated.
```
Comment (nghi-ly) [inline website/docs/reference/events-logging.md]: does it make sense to remove this callout? it reads like a release note entry.

**Review by nghi-ly — COMMENTED** [review]
looks great! left just one nit and one question

File: website/docs/reference/events-logging.md (hunk 2 above)
Comment (matthewshaver) [reply website/docs/reference/events-logging.md]: Good call! I'll remove it

**Review by nghi-ly — APPROVED** [review]
:shipit:

**Pull request merged**